### PR TITLE
Added note about noderestriction labels issue-#31972

### PR DESCRIPTION
**Pull request by PriyanshuAhlawat**
This pr adds a new section titled as-Managed Node Labels , on the page-Creating a cluster with kubeadm.

### CODE REVIEW

File: content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md [hunk 1]
```diff
@@ -284,6 +284,15 @@ If your network is not working or CoreDNS is not in the `Running` state, check o
 [troubleshooting guide](/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm/)
 for `kubeadm`.
 
+### Managed node labels
+
+By default, kubeadm enables the [NodeRestriction](/docs/reference/access-authn-authz/admission-controllers/#noderestriction)
+admission controller that restricts what labels can be self-applied by kubelets on node registration.
+The admission controller documentation covers what labels are permitted to be used with the kubelet `--node-labels` option.
+The `node-role.kubernetes.io/control-plane` label is such a restricted label and kubeadm manually applies it using
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md:292]: https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#noderestriction _does not_ list
`node-role.kubernetes.io/control-plane` as a restricted label @PriyanshuAhlawat 

It is however not on the allowlist, and is a **reserved** label by dint of not being on the allowlist.

**Review by sftim — DISMISSED** [review]
I don't think this change is factually correct.

SIG Architecture folks might be able to confirm authoritatively. I mentioned SIG Architecture because this mention of restriction / lack of restriction cuts across at least 3 SIGs: auth, cluster lifecycle, and node.

File: content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md (hunk 1 above)
Comment (PriyanshuAhlawat) [reply content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md:292]: Should I remove `node-role.kubernetes.io/control-plane` from the pr?

Comment (neolit123) [reply content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md:292]: The relevant part from the link is:
> Use of any other labels under the kubernetes.io or k8s.io prefixes by kubelets is reserved, and may be disallowed or allowed by the NodeRestriction admission plugin in the future.

As such, node-role labels are restricted.
Kubectl and kubeadm are the only relevant users of these. Also core k8s has no concept of a node-role.

**Comment by neolit123** [comment]
> I don't think this change is factually correct

The change is quite accurate.

> SIG Architecture folks might be able to confirm authoritatively. I mentioned SIG Architecture because this mention of restriction / lack of restriction cuts across at least 3 SIGs: auth, cluster lifecycle, and node.

I don't think that is needed. There is some history around node-role labels and they are considered non-core at this point. You can read here:
https://github.com/kubernetes/enhancements/blob/926a1238304eba922be389a99d51c36cda3510ed/keps/sig-architecture/1143-node-role-labels/README.md

The fact they are in the kubernetes.io namespace is the only factual restriction as per the NodeRestriction controller implementation.

**Review by neolit123 — COMMENTED** [review]
/lgtm
/sig cluster-lifecycle
Thanks for the PR @PriyanshuAhlawat

**Comment by sftim** [comment]
This is OK to merge, but I'll open another issue to fix the wording. It's really not clear enough.

/approve

**Comment by sftim** [comment]
Thanks @PriyanshuAhlawat!

PS. I just noticed that the PR title will trigger a Prow warning when we merge this into _dev-1.24_ - I didn't spot this before approving.

**Pull request merged**